src/DataValues/BoundedQuantityValue.php

```diff
@@ -0,0 +1,381 @@
+		$amount = $this->amount->getValueFloat();
+		$upperBound = $this->upperBound->getValueFloat();
+		$lowerBound = $this->lowerBound->getValueFloat();
+		$precision = min( $amount - $lowerBound, $upperBound - $amount );
+
+		if ( $precision === 0.0 ) {
+			// If there is no uncertainty, the order of uncertainty is a bit more than what we have digits for.
+			return -strlen( $this->amount->getFractionalPart() );
+		}
+
+		// e.g. +/- 200 -> 2; +/- 0.02 -> -2
+		// note: we really want floor( log10( $precision ) ), but have to account for
+		// small errors made in the floating point operations above.
+		// @todo: use bcmath (via DecimalMath) to avoid this if possible
+		$orderOfUncertainty = floor( log10( $precision + 0.0000000005 ) );
+
+		return (int)$orderOfUncertainty;
+	}
+
+	/**
+	 * Returns the number of significant figures in the amount-string,
+	 * counting the decimal point, but not counting the leading sign.
+	 *
+	 * Note that this calculation assumes a symmetric uncertainty interval, and can be misleading
+	 *
+	 * @since 0.1
+	 *
+	 * @return int
+	 */
+	public function getSignificantFigures() {
+		$math = new DecimalMath();
+
+		// $orderOfUncertainty is +/- 200 -> 2; +/- 0.02 -> -2
+		$orderOfUncertainty = $this->getOrderOfUncertainty();
+
+		// the number of digits (without the sign) is the same as the position (with the sign).
+		$significantDigits = $math->getPositionForExponent( $orderOfUncertainty, $this->amount );
+
+		return $significantDigits;
+	}
+
+	/**
+	 * Returns a transformed value derived from this QuantityValue by applying
+	 * the given transformation to the amount and the upper and lower bounds.
+	 * The resulting amount and bounds are rounded to the significant number of
+	 * digits. Note that for exact quantities (with at least one bound equal to
+	 * the amount), no rounding is applied (since they are considered to have
+	 * infinite precision).
+	 *
+	 * The transformation is provided as a callback, which must implement a
+	 * monotonously increasing, fully differentiable function mapping a DecimalValue
+	 * to a DecimalValue. Typically, it will be a linear transformation applying a
+	 * factor and an offset.
+	 *
+	 * @param string $newUnit The unit of the transformed quantity.
+	 *
+	 * @param callable $transformation A callback that implements the desired transformation.
+	 *        The transformation will be called three times, once for the amount, once
+	 *        for the lower bound, and once for the upper bound. It must return a DecimalValue.
+	 *        The first parameter passed to $transformation is the DecimalValue to transform
+	 *        In addition, any extra parameters passed to transform() will be passed through
+	 *        to the transformation callback.
+	 *
+	 * @param mixed ... Any extra parameters will be passed to the $transformation function.
+	 *
+	 * @throws InvalidArgumentException
+	 * @return self
+	 */
+	public function transform( $newUnit, $transformation ) {
```

Comment:
We will need it for unit conversion, for formatting and indexing. Deleting tricky code is asking for trouble. I'm open to moving it though, see above.